### PR TITLE
Enchantments, a handful of bug fixes

### PR DESCRIPTION
**Pull request by Psither**
The first pass at having items that have built-in enchantments that can't be removed, but don't block normal enchanting logic

Fixes silk touch giving dragon dust items
Fixes a body type not being assigned properly when using the dragon command without specifying it
Fixes base body type not specifying health bonus

### CODE REVIEW

**Comment by SiverDX** [comment]
is `getDefaultEnchantments` only called on the client?

How is the server going to determine the default enchantments?
Why not pass a level / registry access to the method, don't the events have access to that (level or an entity)?

Also would recommend adding a new method to 
https://github.com/DragonSurvivalTeam/DragonSurvival/blob/1.21.x/src/main/java/by/dragonsurvivalteam/dragonsurvival/util/EnchantmentUtils.java
which gets an enchantment, to cut down on boilerplate code

**Comment by Psither** [comment]
ItemStack doesn't have any way to access the level directly, or any entities to get the level from, and that's all that most of these methods get.

**Comment by SiverDX** [comment]
I mean at the moment it just wouldn't work, would it?
In `applyDebuffs` you have an entity but still access the client minecraft instance
(and iirc effects can only really be modified on the server side)

same for the `livingHurt` event

you could maybe popualte a static list on `ServerStartedEvent` (if the server has registry access, don't know tbh)

File: src/main/java/by/dragonsurvivalteam/dragonsurvival/common/items/armor/DragonHunterSword.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    public DragonHunterSword(Properties pProperties) {
+        super(DSEquipment.DRAGON_HUNTER, pProperties);
+    }
+
+    @Override
+    public ItemEnchantments getDefaultEnchantments() {
+        ItemEnchantments.Mutable temp = new ItemEnchantments.Mutable(ItemEnchantments.EMPTY);
+        if (Minecraft.getInstance().level != null && Minecraft.getInstance().level.registryAccess().registry(Registries.ENCHANTMENT).isPresent()) {
+            temp.set(Minecraft.getInstance().level.registryAccess().registry(Registries.ENCHANTMENT).get()
+                    .getHolderOrThrow(DSEnchantments.DRAGONSBANE), 3);
```
Comment (seelderr) [inline src/main/java/by/dragonsurvivalteam/dragonsurvival/common/items/armor/DragonHunterSword.java:21]: I was thinking about naming a special sword Dragonsbane, but the enchantment I would just call Bane of Dragons to match up with vanilla names

File: src/main/java/by/dragonsurvivalteam/dragonsurvival/mixins/MixinGrindstoneMenu.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(GrindstoneMenu.class)
+public class MixinGrindstoneMenu {
+    @Shadow @Final private Container repairSlots;
+
+    @Inject(method="removeNonCursesFrom(Lnet/minecraft/world/item/ItemStack;)Lnet/minecraft/world/item/ItemStack;", at = @At(value= "RETURN"))
+    private void addDefaultEnchantmentsBack(ItemStack pItem, CallbackInfoReturnable<ItemStack> cir) {
```
Comment (seelderr) [inline src/main/java/by/dragonsurvivalteam/dragonsurvival/mixins/MixinGrindstoneMenu.java:20]: Is this the part where it does remove only non-permanent enchants?

File: src/main/java/by/dragonsurvivalteam/dragonsurvival/registry/DSEnchantments.java [hunk 3]
```diff
@@ -12,8 +12,21 @@ private static ResourceKey<Enchantment> register(String key) {
...
+    public static ResourceKey<Enchantment> MURDERERS_CUNNING = register("murderers_cunning");
+    public static ResourceKey<Enchantment> OVERWHELMING_MIGHT = register("overwhelming_might");
+    public static ResourceKey<Enchantment> DRACONIC_SUPERIORITY = register("draconic_superiority");
+
+    public static ResourceKey<Enchantment> COMBAT_RECOVERY = register("combat_recovery");
+    public static ResourceKey<Enchantment> UNBREAKABLE_SPIRIT = register("unbreakable_spirit");
+    public static ResourceKey<Enchantment> AERODYNAMIC_MASTERY = register("aerodynamic_mastery");
+    public static ResourceKey<Enchantment> SACRED_SCALES = register("sacred_scales");
+
+    public static ResourceKey<Enchantment> CURSE_OF_OUTLAW = register("curse_of_outlaw");
```
Comment (seelderr) [inline src/main/java/by/dragonsurvivalteam/dragonsurvival/registry/DSEnchantments.java:30]: "Curse of Outlaws" ?

**Review by seelderr — CHANGES_REQUESTED** [review]

**Comment by TeamSpen210** [comment]
Instead of directly adding enchantments to NBT, it might be better to instead override [`getEnchantmentLevel()`](https://github.com/neoforged/NeoForge/blob/c636694bd5135d520c7b162d7b38f00cfc583364/src/main/java/net/neoforged/neoforge/common/extensions/IItemExtension.java#L448) and `getAllEnchantments()`. That way the item is considered to  have the enchantment for gameplay purposes, but it can't be removed with the grindstone, regular enchantments can still work, etc.

File: src/main/java/by/dragonsurvivalteam/dragonsurvival/registry/DSEnchantments.java (hunk 3 above)
Comment (Psither) [reply src/main/java/by/dragonsurvivalteam/dragonsurvival/registry/DSEnchantments.java:30]: Placeholder value

**Review by seelderr — APPROVED** [review]

**Pull request merged**